catalog-scripts/register_discovered_locations_bitbucket.py

```diff
@@ -0,0 +1,75 @@
+import requests
+import re
+from requests.auth import HTTPBasicAuth
+from requests.packages.urllib3.util.retry import Retry
+from requests.adapters import HTTPAdapter
+
+username = "" # Your Bitbucket username
+workspace = "" # Your Bitbucket workspace
+app_password = "" # Your Bitbucket app_password
+repository = "" # Your Bitbucket repo
+branch = "main"
+account = "" 
+yaml_filename = "catalog-info.yaml"
+
+# Replace with your x-api-key. Refer https://developer.harness.io/docs/platform/automation/api/api-quickstart/#create-a-harness-api-key-and-token to generate one
+x_api_key = ""
+
+api_url = f"https://idp.harness.io/{account}/idp/api/catalog/locations"
+
+directories = []
+dir_url = f"https://api.bitbucket.org/2.0/repositories/{workspace}/{repository}/src"
+response = requests.get(dir_url, auth=HTTPBasicAuth(username, app_password))
+if response.status_code != 200:
+    print(f"Failed to fetch the latest commit. Status code: {response.status_code}")
+    exit()
+
+def extract_directories(response):
+    return [item["path"] for item in response.json().get("values", []) if re.search(r'-service', item["path"])]
```

Comment:
Not all directories ending with `-service` could be a microservice with catalog-info.yaml
We need to filter it using a regex catalog path like `r"mock_rserver_root/configs/services/.*?/.ownership/catalog-info.yaml".` just like how github discovery plugin does and how we've done in github script

Reply:
Resolved the comment. Please check again